src/basic/path-util.c

```diff
@@ -853,3 +854,24 @@ char *prefix_root(const char *root, const char *path) {
         strcpy(p, path);
         return n;
 }
+
+bool path_prefix_in_list(const char *path, const char *x, ...) {
+        const char *p;
+        va_list ap;
+
+        if (!path || !x)
+                return false;
+
+        va_start(ap, x);
+
+        if (path_startswith(path, x))
+                return true;
+
+        while ((p = va_arg(ap, const char*)))
+                if (path_startswith(path, p))
+                        return true;
+
+        va_end(ap);
```

Comment:
`va_end` needs to be called always, so there are calls missing in the early return cases.